stl/src/syncstream.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// initialize syncstream mutex map
+
+#include <map>
+#include <shared_mutex>
+
+#pragma warning(disable : 4074)
+#pragma init_seg(compiler)
+
+_STD_BEGIN
+// OBJECT DECLARATIONS
+struct _Mutex_count_pair {
+    shared_mutex _Mutex;
+    uint64_t _Ref_count = 0;
+};
+static map<void*, _Mutex_count_pair> _Mutex_map{};
+static shared_mutex _Mutex{};
+
+extern "C" _CRTIMP2 shared_mutex& _Get_mutex_for_instance(void* _Ptr) {
+    scoped_lock _Guard(_Mutex);

```

Comment:
```suggestion
    shared_lock _Guard(_Mutex);
```
`find` and `end` are guaranteed to not modify `map` so this function can be called concurrently.